Lib/test/test_sqlite3/test_regression.py

```diff
@@ -469,6 +469,18 @@ def test_executescript_step_through_select(self):
             con.executescript("select step(t) from t")
             self.assertEqual(steps, values)
 
+    def test_custom_cursor_object_crash_gh_99886(self):
+        # This test segfaults on GH-99886
+        class MyCursor(sqlite.Cursor):
+            def __init__(self, *args, **kwargs):
+                super().__init__(*args, **kwargs)
+                # this can go before or after the super call; doesn't matter
+                self.some_attr = None
+
+        conn = sqlite.connect(':memory:')
+        cur = conn.cursor(MyCursor)
+        cur.close()
+        del cur
```

Comment:
We can use the `memory_database()` helper to create and destroy a dummy `:memory:` database connection:

```suggestion
        with memory_database() as con:
            cur = con.cursor(MyCursor)
            cur.close()
            del cur
```